src/librustc/hir/mod.rs

```diff
@@ -1837,7 +1837,7 @@ pub enum ImplItemKind {
     /// A method implementation with the given signature and body.
     Method(MethodSig, BodyId),
     /// An associated type.
-    Type(P<Ty>),
+    TyAlias(P<Ty>),
```

Comment:
Basically I think this should be reverted. Also, as per some issue, `Method` should be `Fn`.